### ISSUE TITLE
Mappings from another collection do not work properly

### ISSUE THREAD

**Issue by maxxxxxdlp** [specify/specify6#694]
TL;DR: mappings created in other collections are not shown in DataExporter. Also, they can not be exported from another collection

After creating a mapping (let's call it `m1`) in a collection 1 (which does not have any other mapping), it was displayed normally in Specify and DataExporter

In collection 2 (which has some other mappings) this mapping is hidden behind the `Other Queries` button. Pressing it opens the mapping normally (Unless the `Overwrite` windows appear, in which case you can not open it)
<img width="286" alt="Screen Shot 2020-03-25 at 2 00 59 PM" src="https://user-images.githubusercontent.com/40512816/77575723-f8bac080-6ea1-11ea-83ad-8e4d404f197e.png">

Yet, **DataExporter from collection 2 does not show the mapping created in collection 1**
<img width="703" alt="Screen Shot 2020-03-25 at 1 59 33 PM" src="https://user-images.githubusercontent.com/40512816/77575721-f8222a00-6ea1-11ea-9259-8dad7edd1f3b.png">

Deleting all of the mappings from 2 did not cause the `m1` to reappear in the list as it still was hidden behind the `Other Queries`
Creating copies of `m1` (by pressing `Save As`) worked well and the copies were displayed normally in 2
<img width="286" alt="Screen Shot 2020-03-25 at 2 05 01 PM" src="https://user-images.githubusercontent.com/40512816/77575729-fa848400-6ea1-11ea-9358-3928654bfadf.png">

But they appeared only under `Other Queries` in 1
<img width="565" alt="Screen Shot 2020-03-25 at 2 23 15 PM" src="https://user-images.githubusercontent.com/40512816/77577157-4c2e0e00-6ea4-11ea-8f76-daa228dbbd7b.png">

The copies appear normally in `DataExporter` but `m1` does not
<img width="703" alt="Screen Shot 2020-03-25 at 2 17 54 PM" src="https://user-images.githubusercontent.com/40512816/77576580-6e735c00-6ea3-11ea-8cb9-600cbc43a834.png">

When trying to export `m1` from 2, there is an option `Other Queries` in the list
<img width="317" alt="Screen Shot 2020-03-25 at 2 05 12 PM" src="https://user-images.githubusercontent.com/40512816/77575731-fb1d1a80-6ea1-11ea-96bf-8eb81e0e9b85.png">

Selecting it generates an almost empty file:
<img width="744" alt="Screen Shot 2020-03-25 at 2 13 33 PM" src="https://user-images.githubusercontent.com/40512816/77576225-d2495500-6ea2-11ea-9093-9b4a65454316.png">

Also, when exporting queries in 1, all of the coppies of `m1` were "summarized" under one option `Other Queries`
<img width="317" alt="Screen Shot 2020-03-25 at 2 23 42 PM" src="https://user-images.githubusercontent.com/40512816/77577235-68ca4600-6ea4-11ea-970b-5f6bd93219df.png">

All of the things done in 2 could also be replicated in 3

---

See also #693

**Comment by norinespears** [comment]
I saw strange behavior in the Schema Mapper as well. Here are my steps:

1)In Collection 1 create new mapping and save as New Mapping
2)Leave Collection 1 and open Collection 2
3)Open Schema Mapper
The new schema is not present in the sidebar which I think is to be expected?
***I need to ask you about this i.e. do I need to export and import the mapping? I tried to do that but I can not import the mapping ... 
4)There an 'other queries' in the sidebar and when I click on it it is in fact the same query I used to create my New Mapping in Collection 1 (not sure if this is expected?)
So, I save that query as a New Mapping
When I do this the sidebar goes a bit wacky and shows Save Queries -- although my New Mapping is also present. I'll put screenshots and log file in a new comment
5) I open Collection 1 and 'Other Queries' is now on that sidebar as well
6) I open Collection 2 and the sidebar is now back to it's original state with New Mapping included on the sidebar, Other Queries also present but Queries no longer present

I'll put screenshots and the log file in another comment

**Comment by norinespears** [comment]
Here is the log file just in case it is useful:
************************ Type: App  Action: StartUp************************ Type: App  Action: StartUp
Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]
Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
Couldn't find the ViewDef for formtable definition name[Collectors]
Couldn't find the ViewDef for formtable definition name[Collectors]Couldn't find the ViewDef for formtable definition name[Collectors]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
************************ Type: App  Action: AppRestart
************************ Type: App  Action: AppRestart************************ Type: App  Action: AppRestart
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
Couldn't find the ViewDef for formtable definition name[Collectors]
Couldn't find the ViewDef for formtable definition name[Collectors]Couldn't find the ViewDef for formtable definition name[Collectors]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
************************ Type: App  Action: AppRestart
************************ Type: App  Action: AppRestart************************ Type: App  Action: AppRestart
Exception in thread "AWT-EventQueue-0" Exception in thread "AWT-EventQueue-0" java.lang.NullPointerException
java.lang.NullPointerExceptionjava.lang.NullPointerException
	at edu.ku.brc.specify.tasks.subpane.qb.QueryBldrPane.saveQuery(QueryBldrPane.java:3417)
	at edu.ku.brc.specify.tasks.subpane.qb.QueryBldrPane.saveQuery(QueryBldrPane.java:3417)	at edu.ku.brc.specify.tasks.subpane.qb.QueryBldrPane.saveQuery(QueryBldrPane.java:3417)
	at edu.ku.brc.specify.tasks.subpane.qb.QueryBldrPane$2.actionPerformed(QueryBldrPane.java:508)
	at edu.ku.brc.specify.tasks.subpane.qb.QueryBldrPane$2.actionPerformed(QueryBldrPane.java:508)	at edu.ku.brc.specify.tasks.subpane.qb.QueryBldrPane$2.actionPerformed(QueryBldrPane.java:508)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)	at javax.swing.AbstractButton.doClick(AbstractButton.java:376)
	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:842)
	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:842)	at javax.swing.plaf.basic.BasicMenuItemUI.doClick(BasicMenuItemUI.java:842)
	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:886)
	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:886)	at javax.swing.plaf.basic.BasicMenuItemUI$Handler.mouseReleased(BasicMenuItemUI.java:886)
	at java.awt.Component.processMouseEvent(Component.java:6539)
	at java.awt.Component.processMouseEvent(Component.java:6539)	at java.awt.Component.processMouseEvent(Component.java:6539)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6304)
	at java.awt.Component.processEvent(Component.java:6304)	at java.awt.Component.processEvent(Component.java:6304)
	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Container.processEvent(Container.java:2239)	at java.awt.Container.processEvent(Container.java:2239)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Component.dispatchEventImpl(Component.java:4889)	at java.awt.Component.dispatchEventImpl(Component.java:4889)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Container.dispatchEventImpl(Container.java:2297)	at java.awt.Container.dispatchEventImpl(Container.java:2297)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.Component.dispatchEvent(Component.java:4711)	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4904)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4535)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4476)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Container.dispatchEventImpl(Container.java:2283)	at java.awt.Container.dispatchEventImpl(Container.java:2283)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.Component.dispatchEvent(Component.java:4711)	at java.awt.Component.dispatchEvent(Component.java:4711)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:760)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue.access$500(EventQueue.java:97)	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:709)	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.awt.EventQueue$3.run(EventQueue.java:703)	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessController.doPrivileged(Native Method)	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:84)
	at java.awt.EventQueue$4.run(EventQueue.java:733)
	at java.awt.EventQueue$4.run(EventQueue.java:733)	at java.awt.EventQueue$4.run(EventQueue.java:733)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:731)	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.AccessController.doPrivileged(Native Method)	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)	at java.awt.EventQueue.dispatchEvent(EventQueue.java:730)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]
Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
Couldn't find the ViewDef for formtable definition name[Collectors]
Couldn't find the ViewDef for formtable definition name[Collectors]Couldn't find the ViewDef for formtable definition name[Collectors]
************************ Type: App  Action: AppRestart
************************ Type: App  Action: AppRestart************************ Type: App  Action: AppRestart
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
Couldn't find the ViewDef for formtable definition name[Collectors]
Couldn't find the ViewDef for formtable definition name[Collectors]Couldn't find the ViewDef for formtable definition name[Collectors]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
************************ Type: App  Action: AppRestart
************************ Type: App  Action: AppRestart************************ Type: App  Action: AppRestart
Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]
Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
Couldn't find the ViewDef for formtable definition name[Collectors]
Couldn't find the ViewDef for formtable definition name[Collectors]Couldn't find the ViewDef for formtable definition name[Collectors]
************************ Type: App  Action: AppRestart
************************ Type: App  Action: AppRestart************************ Type: App  Action: AppRestart
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
Couldn't find the ViewDef for formtable definition name[Collectors]
Couldn't find the ViewDef for formtable definition name[Collectors]Couldn't find the ViewDef for formtable definition name[Collectors]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
************************ Type: App  Action: AppRestart
************************ Type: App  Action: AppRestart************************ Type: App  Action: AppRestart
Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]
Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]Setting Label - Form control with id[1] is not in ViewDef or Panel [Preparation] in ViewSet [Botany Views]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
Couldn't find the ViewDef for formtable definition name[Collectors]
Couldn't find the ViewDef for formtable definition name[Collectors]Couldn't find the ViewDef for formtable definition name[Collectors]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
************************ Type: App  Action: AppRestart
************************ Type: App  Action: AppRestart************************ Type: App  Action: AppRestart
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
Couldn't find the ViewDef for formtable definition name[Collectors]
Couldn't find the ViewDef for formtable definition name[Collectors]Couldn't find the ViewDef for formtable definition name[Collectors]
************************ Type: App  Action: AppRestart
************************ Type: App  Action: AppRestart************************ Type: App  Action: AppRestart
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]The Field Name [pcrPerson] was not in the Table [DNA Sequence] in ViewSet [Common]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]The Field Name [doSearch] was not in the Table [Collection Object] in ViewSet [Search]
Couldn't find the ViewDef for formtable definition name[Collectors]
Couldn't find the ViewDef for formtable definition name[Collectors]Couldn't find the ViewDef for formtable definition name[Collectors]
************************ Type: App  Action: AppRestart
************************ Type: App  Action: AppRestart************************ Type: App  Action: AppRestart

**Comment by timo11** [comment]
Other Queries are banished from the Schema Mapper sidebar. I don't really see how they could have caused the problem with imported mappings not showing up in the exporter, but I was unable to duplicate that problem. Hopefully it is fixed.
